Inference Rules Updates for multiple Nodes (#265)
mlcommons/inference_policies: inference_rules.adoc

* The QDL is not allowed to pad the data in queries.
* The QDL is not allowed to cache queries or responses.
* The QDL is implementing the network function of the LoadGen Node towards the SUT node and handles the required processing. E.G. padding of the payload as required by the network protocol.

> * The QDL should reflect a single SUT to the LoadGen. LoadGen Operates with a single SUT. 

Reviewer: ```suggestion
* The QDL should reflect a single SUT to the LoadGen. LoadGen operates with a single SUT. 
```

Author: changed

* The Name method's return value must contain the substring "Network SUT".
* The Name method's implementation must include at least one round trip over the network. The Name method must not return until the round trip is complete.

> * The QDL should serially get the names from all the Nodes of the SUT, aggregate the response in the Name Method after getting response from all Nodes. Each Node must have a unique Name.   

Reviewer: ```suggestion
* The QDL should serially get the names from all the Nodes of the SUT and aggregate the responses in the Name Method after getting response from all Nodes. Each Node must have a unique Name.   
```

Author: changed

> * The QDL should serially get the names from all the Nodes of the SUT and aggregate the responses in the Name Method after getting response from all Nodes. Each Node must have a unique Name.   

Reviewer: ```suggestion
* The QDL must query each SUT Node for its name and aggregate the responses in the Name Method. Each SUT Node must have a unique name.   
```

Author: One clarification please - the word serially is not appearing in your text. Do we care if in multiple nodes we query in parallel? Do we consider the latency accumulation as an additional information or just want to see it is Network > local?

Author: OK I accept your suggestion

Additionally, any interconnect not listed in the permitted list is forbidden unless clearance is first obtained from the MLPerf Inference WG.

> The SUT and QDL must run on two physically separated and distinct systems. The SUT can contain multiple Nodes.

Reviewer: ```suggestion
The SUT and QDL must run on physically separate and distinct systems. The SUT can contain multiple Nodes.
```

Author: thanks

Fabric and protocol must be reported in the submission metadata. Submission metadata must be sufficient to determine OSI layers one through four of the submission's network stack.

==== SUT over the Network and Setup Constraints

* SUT parameters and configuration must be uniquely and specifically named in the submission results.

> * Everything outside the LoadGen node should be considered as part of the SUT, for instance for counting power and latency. As an example, components outside the nodes like switch or load balancer should be considered part of the SUT. The SUT can contain multiple Nodes.

Reviewer: ```suggestion
* Everything outside the LoadGen node should be considered as part of the SUT, for instance for counting power and latency. As an example, components outside the nodes like a switch or load balancer should be considered part of the SUT. The SUT can contain multiple Nodes.
```

Author: changed

> * Everything outside the LoadGen node should be considered as part of the SUT, for instance for counting power and latency. As an example, components outside the nodes like a switch or load balancer should be considered part of the SUT. The SUT can contain multiple Nodes.

Reviewer: Just for "don't repeat yourself" reasons I suggest we just say that the SUT can contain multiple nodes one time.

Author: OK. I will delete it here.

* All queries must be transferred over the network, carrying the inference data, for inference execution at the SUT. All responses must be transferred back over the network, carrying the inference responses. 
* Caching/Storing of the queries and inference data or responses for further use at the SUT is disallowed. It is allowed to cache/store other data like Neural Network weights or Neural Network executable.
* SUT can do the required pre-processing of the data, e.g. Batching, Padding, processing of the requests (precision, data layout), compression, decompression. SUT can do the required post processing functions e.g. gather, reduction or ArgMax. 

> * Network interface characteristics must be added to the report, both at the SUT side, for every Node, and at the LoadGen system side, including networking components, and name any other components through which data passes between LoadGen and the SUT for reproducibility.

Reviewer: ```suggestion
* Network interface characteristics must be added to the report, both at the SUT side for every Node, and at the LoadGen system side, including networking components. Every other component through which data passes between LoadGen and the SUT must also be described for reproducibility purposes.
```

Author: changed

> * Network interface characteristics must be added to the report, both at the SUT side for every Node, and at the LoadGen system side, including networking components.  Every other component through which data passes between LoadGen and the SUT must also be described for reproducibility purposes.

Reviewer: ```suggestion
* The report must contain network interface characteristics for both the Loadgen and SUT systems, and every other component through which data passes between Loadgen and SUT. The information must be sufficient for reproducibility.
```

Author: thanks

> * It is required to add a system diagram that shows how the components between the LoadGen node and the SUT nodes are connected and to add text that explains the diagram.

Reviewer: ```suggestion
* A system diagram must be included in the submission that shows how the components between the LoadGen node and the SUT nodes are connected, accompanied by any text necessary for another submitter to understand the diagram.
```

Author: changed
